src/inference/src/dev/core_impl.cpp

```diff
@@ -1447,6 +1447,19 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
 
                 ov::AnyMap update_config = config;
                 update_config[ov::loaded_from_cache.name()] = true;
+
+                if (update_config[ov::cache_mode.name()] == ov::CacheMode::OPTIMIZE_SIZE) {
```

Comment:
cache mode can be globally set before using `core.set_property(GPU, cache_mode(OPTIMIZE_SIZE))` and not present in current `config` / `update_config`

Reply:
probably, we can unconditionally set weight_path to all plugins and just add checks in each plugin to ignore this property if it's not handled by plugin

Reply:
Done. Makes sense, if OPTIMIZE_SIZE is not set, weights_path will be ignored even if it's set here.